presto-main/src/test/java/com/facebook/presto/cost/TestCapDistinctValuesCountToTypeDomainRangeLength.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.cost;
+
+import com.facebook.presto.block.BlockEncodingManager;
+import com.facebook.presto.metadata.FunctionRegistry;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.TypeManager;
+import com.facebook.presto.sql.analyzer.FeaturesConfig;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.testing.TestingConnectorSession;
+import com.facebook.presto.type.TypeRegistry;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.SmallintType.SMALLINT;
+import static com.facebook.presto.spi.type.TinyintType.TINYINT;
+import static java.util.Collections.emptyList;
+import static org.testng.Assert.assertEquals;
+
+public class TestCapDistinctValuesCountToTypeDomainRangeLength
+{
+    private final TypeManager typeManager = new TypeRegistry();
+    private final FunctionRegistry functionRegistry = new FunctionRegistry(typeManager, new BlockEncodingManager(typeManager), new FeaturesConfig());
+    private final ConnectorSession session = new TestingConnectorSession(emptyList());
+
+    @Test
+    public void test()
```

Comment:
Add case for DECIMAL (both Integral and not integral).
Add case for non integral type (eg. DOUBLE).
Add case for integral type that will have stats unaffected by normalizer (range > ndv)